File: shared/main/java/net/ccbluex/liquidbounce/utils/ClientUtils.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.utils;

import java.security.PublicKey;

import javax.crypto.SecretKey;

import com.google.gson.JsonObject;

import net.ccbluex.liquidbounce.LiquidBounce;
import net.ccbluex.liquidbounce.api.minecraft.INetworkManager;
import net.ccbluex.liquidbounce.api.minecraft.network.login.server.ISPacketEncryptionRequest;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

@SideOnly(Side.CLIENT)
public final class ClientUtils extends MinecraftInstance
{

	private static final Logger logger = LogManager.getLogger("LiquidBounce");

	public static Logger getLogger()
	{
		return logger;
	}

	public static void disableFastRender()
	{
		LiquidBounce.wrapper.getFunctions().disableFastRender();
	}

	public static void sendEncryption(final INetworkManager networkManager, final SecretKey secretKey, final PublicKey publicKey, final ISPacketEncryptionRequest encryptionRequest)
	{
		networkManager.sendPacket(classProvider.createCPacketEncryptionResponse(secretKey, publicKey, encryptionRequest.getVerifyToken()), () ->
		{
			networkManager.enableEncryption(secretKey);

			return null;
		});
	}

	public static void displayChatMessage(final String message)
	{
		if (mc.getThePlayer() == null)
		{
			logger.info("(MCChat){}", message);
			return;
		}

		final JsonObject jsonObject = new JsonObject();
		jsonObject.addProperty("text", message);

		mc.getThePlayer().addChatMessage(LiquidBounce.wrapper.getFunctions().jsonToComponent(jsonObject.toString()));
	}
}

File: shared/main/java/net/ccbluex/liquidbounce/utils/ClientUtils.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.utils

import com.google.gson.JsonObject
import net.ccbluex.liquidbounce.LiquidBounce
import net.ccbluex.liquidbounce.LiquidBounce.wrapper
import net.ccbluex.liquidbounce.api.minecraft.INetworkManager
import net.ccbluex.liquidbounce.api.minecraft.client.entity.IEntityPlayerSP
import net.ccbluex.liquidbounce.api.minecraft.network.login.server.ISPacketEncryptionRequest
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import java.security.PublicKey
import javax.crypto.SecretKey

@SideOnly(Side.CLIENT)
object ClientUtils : MinecraftInstance()
{
	@JvmStatic
	val logger: Logger = LogManager.getLogger(LiquidBounce.CLIENT_NAME)

	@JvmStatic
	fun disableFastRender()
	{
		wrapper.functions.disableFastRender()
	}

	@JvmStatic
	fun sendEncryption(networkManager: INetworkManager, secretKey: SecretKey, publicKey: PublicKey, encryptionRequest: ISPacketEncryptionRequest)
	{
		networkManager.sendPacket(classProvider.createCPacketEncryptionResponse(secretKey, publicKey, encryptionRequest.verifyToken)) { networkManager.enableEncryption(secretKey) }
	}

	@JvmStatic
	fun displayChatMessage(thePlayer: IEntityPlayerSP?, message: String?)
	{
		if (thePlayer == null)
		{
			logger.info("(MCChat) {}", message)
			return
		}

		val jsonObject = JsonObject()
		jsonObject.addProperty("text", message)
		thePlayer.addChatMessage(wrapper.functions.jsonToComponent("$jsonObject"))
	}
}
